provider/hybrid.go

```diff
@@ -66,7 +69,139 @@ func (dp dockerHybridProvider) CheckConfig(ctx context.Context, request *rpc.Che
 
 func (dp dockerHybridProvider) DiffConfig(ctx context.Context, request *rpc.DiffRequest) (*rpc.DiffResponse, error) {
 	// Delegate to the bridged provider, as native Provider does not implement it.
-	return dp.bridgedProvider.DiffConfig(ctx, request)
+
+	urn := resource.URN(request.GetUrn())
+	label := fmt.Sprintf("DiffConfig(%s)", urn)
+
+	olds, err := plugin.UnmarshalProperties(request.GetOlds(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.olds", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	news, err := plugin.UnmarshalProperties(request.GetNews(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.news", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("DiffConfig failed because of malformed resource inputs: %w", err)
+	}
+
+	// forcing this to true for the prototype. This needs to be configurable
+	var ignoreRegistryAuthChanges bool = true
+	if v := news["ignoreRegistryAuthChanges"]; v.HasValue() && v.IsBool() {
+		ignoreRegistryAuthChanges = v.BoolValue()
+	}
+
+	resp, err := dp.bridgedProvider.DiffConfig(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	if hasDiff, err := hasVolatileRegistryAuthDiff(olds, news); err == nil && hasDiff && ignoreRegistryAuthChanges {
+		for idx, prop := range resp.Diffs {
+			if prop == "registryAuth" {
+				resp.Diffs = append(resp.Diffs[:idx], resp.Diffs[idx+1:]...)
+				break
+			}
+		}
+		delete(resp.DetailedDiff, "registryAuth")
+
+		// Set diff to none if there are no other diffs
+		if len(resp.DetailedDiff) == 0 && len(resp.Diffs) == 0 {
+			resp.Changes = rpc.DiffResponse_DIFF_NONE
+		}
+	} else if err != nil {
+		return nil, fmt.Errorf("DiffConfig failed because of malformed registryAuth inputs: %w", err)
+	}
+
+	return resp, err
+}
+
+// hasVolatileRegistryAuthDiff checks if the registryAuth field has a diff that is volatile and should be ignored.
+func hasVolatileRegistryAuthDiff(olds, news resource.PropertyMap) (bool, error) {
+	if !olds.HasValue("registryAuth") || !news.HasValue("registryAuth") {
+		// if old or new registryAuth is missing then there is a diff, no need to clean
+		return false, nil
+	} else if !olds["registryAuth"].IsString() || !news["registryAuth"].IsString() {
+		// if old or new registryAuth is not a string then there is no diff to be cleaned
+		return false, nil
+	} else if olds["registryAuth"].StringValue() == news["registryAuth"].StringValue() {
+		// if the strings are equal then there is no diff to be cleaned
+		return false, nil
+	}
+
+	var oldAuthData []map[string]interface{}
+	var newAuthData []map[string]interface{}
+	
+	err := json.Unmarshal([]byte(olds["registryAuth"].StringValue()), &oldAuthData)
+	if err != nil {
+		return false, err
+	}
+	err = json.Unmarshal([]byte(news["registryAuth"].StringValue()), &newAuthData)
+	if err != nil {
+		return false, err
+	}
```

Comment:
Ugh... IIRC I marked https://github.com/pulumi/pulumi-docker/pull/986 as blocked on https://github.com/pulumi/pulumi/pull/15032 because I really, really didn't want to propagate this pattern of ingesting provider config as JSON blobs.

Reply:
Oh, I must've missed that PR when I had a look at the issue. Thanks for pointing me into that direction!

pulumi/pulumi#15032 definitely seems like the better approach here. With the yaml provider behaving differently then the others (TIL!) this would end up being quite messy with the serialized json..

Do we have plans for merging that in soon? We wanted to tackle pulumi/pulumi-awsx#1203 this iteration and that depends on not having this perma-diff. But it's not worth adding even more cruft just to get it out as soon as possible IMO